src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSTracingPipelineCustomizer.cs

```diff
@@ -31,6 +31,6 @@ public void Customize(Type serviceClientType, RuntimePipeline pipeline)
             return;
         }
 
-        pipeline.AddHandlerBefore<RetryHandler>(new AWSTracingPipelineHandler(this.options));
+        pipeline.AddHandlerBefore<Marshaller>(new AWSTracingPipelineHandler(this.options));
```

Comment:
This will break the `SnsRequestContextHelper` helper which is working on the wire protocol level but now you have injected the handler before that data is setup.

Reply:
Updated `SnsRequestContextHelper` to mirror `SqsRequestContextHelper`